server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/MailboxGetMethod.scala

```diff
@@ -0,0 +1,89 @@
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.method
+
+import eu.timepit.refined.auto._
+import javax.inject.Inject
+import org.apache.james.jmap.json.Serializer
+import org.apache.james.jmap.mail._
+import org.apache.james.jmap.model.Invocation.{Arguments, MethodName}
+import org.apache.james.jmap.model.State.INSTANCE
+import org.apache.james.jmap.model.{Invocation, MailboxFactory}
+import org.apache.james.jmap.utils.quotas.QuotaLoaderWithPreloadedDefaultFactory
+import org.apache.james.mailbox.model.MailboxMetaData
+import org.apache.james.mailbox.model.search.MailboxQuery
+import org.apache.james.mailbox.{MailboxManager, MailboxSession}
+import org.apache.james.metrics.api.MetricFactory
+import org.reactivestreams.Publisher
+import play.api.libs.json.{JsError, JsObject, JsSuccess}
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+class MailboxGetMethod @Inject() (serializer: Serializer,
+                                  mailboxManager: MailboxManager,
+                                  quotaFactory : QuotaLoaderWithPreloadedDefaultFactory,
+                                  mailboxFactory: MailboxFactory,
+                                  metricFactory: MetricFactory) extends Method {
+  override val methodName: MethodName = MethodName("Mailbox/get")
+
+  override def process(invocation: Invocation, mailboxSession: MailboxSession): Publisher[Invocation] = {
+    metricFactory.runPublishingTimerMetricLogP99(JMAP_RFC8621_PREFIX + methodName.value,
+      asMailboxGetRequest(invocation.arguments)
+        .flatMap(mailboxGetRequest => getMailboxes(mailboxGetRequest, mailboxSession)
+          .collectSeq()
+          .map(mailboxes => MailboxGetResponse(
+            accountId = mailboxGetRequest.accountId,
+            state = INSTANCE,
+            list = mailboxes.toList,
+            notFound = NotFound(Nil)))
+          .map(mailboxGetResponse => Invocation(
+            methodName = methodName,
+            arguments = Arguments(serializer.serialize(mailboxGetResponse).as[JsObject]),
+            methodCallId = invocation.methodCallId))))
+  }
+
+  private def asMailboxGetRequest(arguments: Arguments): SMono[MailboxGetRequest] = {
+    serializer.deserializeMailboxGetRequest(arguments.value) match {
+      case JsSuccess(mailboxGetRequest, _) => SMono.just(mailboxGetRequest)
+      case errors: JsError => SMono.raiseError(new IllegalArgumentException(serializer.serialize(errors).toString))
```

Comment:
We should not reuse `IllegalArgumentException` for errors we want to show to the user. It's usually thrown by any code that has preconditions so you can't know you are the one throwing it at http edge